### PR TITLE
Use "note" instead of "notes" in API re dag run / task instance notes

### PR DESCRIPTION
**Pull request by dstandish**
This is a followup to https://github.com/apache/airflow/pull/27849, and it leaves just a tiny bit of room for adding multiple notes in the future (and is _marginally_ more accurate in the meantime)

depends  on https://github.com/apache/airflow/pull/27849 (assumes that merged first! meantime this will be ugly!)

### CODE REVIEW

**Review by jedcunningham — APPROVED** [review]

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -313,6 +313,17 @@ def key(self) -> TaskInstanceKey:
         return self
 
 
+def _creator_note(val):
+    """Custom creator for the ``note`` association proxy."""
+    # breakpoint()
```
Comment (pierrejeambrun) [inline airflow/models/taskinstance.py]: Vestige of debugging session I guess :p

File: airflow/models/dagrun.py [hunk 2]
```diff
@@ -85,6 +87,17 @@ class TISchedulingDecision(NamedTuple):
     finished_tis: list[TI]
 
 
+def _creator_note(val):
+    """Custom creator for the ``note`` association proxy."""
+    # breakpoint()
```
Comment (pierrejeambrun) [inline airflow/models/dagrun.py]: Same for the breakpoint comment. Should the name be reversed _note_creator ?

File: airflow/models/dagrun.py [hunk 3]
```diff
@@ -1295,3 +1307,41 @@ def get_log_filename_template(self, *, session: Session = NEW_SESSION) -> str:
...
+            ondelete="CASCADE",
+        ),
+        ForeignKeyConstraint(
+            (user_id,),
+            ["ab_user.id"],
+            name="dag_run_note_user_fkey",
+        ),
+    )
+
+    def __init__(self, content, user_id=None):
```
Comment (pierrejeambrun) [inline airflow/models/dagrun.py]: Just wondering, why do we need a custom constructor here?

**Review by pierrejeambrun — COMMENTED** [review]
I am not at home so couldn’t do an in depth review or test the code locally but this is looking good 👍

File: airflow/models/dagrun.py (hunk 2 above)
Comment (dstandish) [reply airflow/models/dagrun.py]: `<general>_<specific>`

File: airflow/models/dagrun.py (hunk 3 above)
Comment (dstandish) [reply airflow/models/dagrun.py]: Required for creator

**Review by ashb — APPROVED** [review]

**Pull request merged**